docs/root/start/sandboxes/skywalking_tracing.rst

```diff
@@ -0,0 +1,92 @@
+by the SkyWalking tracer to a SkyWalking cluster (trace driver setup
+in :repo:`/examples/skywalking-tracing/front-envoy-skywalking.yaml`).
+
+When service1 accepts the request forwarded from front envoy, it will makes an API call to service2 before
+returning a response.
+
+Running the Sandbox
+~~~~~~~~~~~~~~~~~~~
+
+.. include:: _include/docker-env-setup.rst
+
+Step 3: Build the sandbox
+*************************
+
+To build this sandbox example, and start the example apps run the following commands:
+
+.. code-block:: console
+
+    $ pwd
+    envoy/examples/skywalking-tracing
+    $ docker-compose pull
+    $ docker-compose up --build -d
+    $ docker-compose ps
+
+                Name                              Command                State                                  Ports
+    --------------------------------------------------------------------------------------------------------------------------------------------------
+    skywalking-tracing_elasticsearch_1    /tini -- /usr/local/bin/do ... Up (healthy)   0.0.0.0:9200->9200/tcp, 9300/tcp
+    skywalking-tracing_front-envoy_1      /docker-entrypoint.sh /bin ... Up             10000/tcp, 0.0.0.0:8000->8000/tcp, 0.0.0.0:8001->8001/tcp
+    skywalking-tracing_service1_1         /bin/sh /usr/local/bin/sta ... Up             10000/tcp
+    skywalking-tracing_service2_1         /bin/sh /usr/local/bin/sta ... Up             10000/tcp
+    skywalking-tracing_skywalking-oap_1   bash docker-entrypoint.sh      Up (healthy)   0.0.0.0:11800->11800/tcp, 1234/tcp, 0.0.0.0:12800->12800/tcp
+    skywalking-tracing_skywalking-ui_1    bash docker-entrypoint.sh      Up             0.0.0.0:8080->8080/tcp
+
+Step 4: Generate some load
+**************************
+
+You can now send a request to service1 via the front-envoy as follows:
+
+.. code-block:: console
+
+    $ curl -v localhost:8000/trace/1
+    *   Trying ::1...
+    * TCP_NODELAY set
+    * Connected to localhost (::1) port 8000 (#0)
+    > GET /trace/1 HTTP/1.1
+    > Host: localhost:8000
+    > User-Agent: curl/7.58.0
+    > Accept: */*
+    >
+    < HTTP/1.1 200 OK
+    < content-type: text/html; charset=utf-8
+    < content-length: 89
+    < server: envoy
+    < date: Sat, 10 Oct 2020 01:56:08 GMT
+    < x-envoy-upstream-service-time: 27
+    <
+    Hello from behind Envoy (service 1)! hostname: 1a2ba43d6d84 resolvedhostname: 172.19.0.6
+    * Connection #0 to host localhost left intact
+
+You can get SkyWalking stats of front-envoy after some requests as follows:
+
+.. code-block:: console
+
+    $ curl -s localhost:8001/stats | grep tracing.skywalking
+    tracing.skywalking.cache_flushed: 0
+    tracing.skywalking.segments_dropped: 0
+    tracing.skywalking.segments_flushed: 0
+    tracing.skywalking.segments_sent: 13
+
+Step 5: View the traces in SkyWalking UI
+****************************************
+
+Point your browser to http://localhost:8080 . You should see the SkyWalking dashboard.
```

Comment:
this needs a test both for http://localhost:8080 and to reflect the subsequent (js/http) call that the browser makes

Reply:
also im afraid it still doesnt work when i run it locally

still the same error

```
Caused by: java.net.UnknownHostException: skywalking-oap: Name does not resolve

```

Reply:
k, i see what is happening

when `skywalking-oap` starts, if elasticsearch is not available then it dies

that is why `skywalking-oap` is not resolving - the container is dead

Reply:
if i try to start `skywalking-oap` after elasticsearch is up im getting a different error

```
skywalking-oap_1  | 2020-10-27 07:38:08,699 - org.apache.skywalking.oap.server.starter.OAPServerBootstrap -38441 [main] ERROR [] - Elasticsearch exception [type=cluster_block_exception, reason=index [ui_template] blocked by: [TOO_MANY_REQUESTS/12/disk usage exceeded flood-stage watermark, index has read-only-allow-delete block];]

```

Reply:
hmm, actually this last error appears to be low disk space on my setup

my guess is that if you can make `skywalking-oap` wait for elasticsearch to become ready before starting, it will work

Reply:
...personally i would just add a bash script to wait - but dockerize is also an option i think https://github.com/jwilder/dockerize#waiting-for-other-dependencies

Reply:
weird. Looks it's OK in CI and my local computer. I will try reproduce it.

Reply:
its not tested in CI - at least not in verify.sh

Reply:
if you look here https://dev.azure.com/cncf/envoy/_build/results?buildId=55574&view=logs&j=f1a49366-51c0-540f-0294-f73c2bb311a9&t=7fd036c9-61ec-5b59-3645-e90950dc67a6&l=7635

youll see its not actually started - if you checked after the delay it should show that it failed

one way of seeing this is using DELAY - ill comment inline...

Reply:
Yes, you are right. But it's not always wrong. I tried three times, two failed and one started properly. We tried to fix it as you suggested.

Reply:
the important thing is adding tests https://github.com/envoyproxy/envoy/pull/13060#discussion_r512473705

Reply:
also my guess - when it passes the composition is already up

if you are testing locally - probs best to do

```
docker-compose down
./verify.sh
```

to be sure

Reply:
👌